pkg/controller/template/render.go

```diff
@@ -260,23 +261,32 @@ func transpileToIgn(files, units []string) (*igntypes.Config, error) {
 
 	// Convert data to Ignition resources
 	for _, d := range files {
-		f := new(igntypes.File)
-		if err := yaml.Unmarshal([]byte(d), f); err != nil {
+		var f igntypes.File
+		if err := yaml.Unmarshal([]byte(d), &f); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal file into struct: %v", err)
 		}
+		regdu := dataurl.New([]byte(*f.Contents.Source), "text/plain")
+		regdu.Encoding = dataurl.EncodingASCII
+		regduString := regdu.String()
+		f.Contents.Source = &regduString
 
 		// Add the file to the config
-		ignCfg.Storage.Files = append(ignCfg.Storage.Files, *f)
+		ignCfg.Storage.Files = append(ignCfg.Storage.Files, f)
 	}
 
 	for _, d := range units {
-		u := new(igntypes.Unit)
-		if err := yaml.Unmarshal([]byte(d), u); err != nil {
+		var u igntypes.Unit
+		if err := yaml.Unmarshal([]byte(d), &u); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal systemd unit into struct: %v", err)
 		}
 
+		regdu := dataurl.New([]byte(*u.Contents), "text/plain")
```

Comment:
all this needs to be dropped, units don't need dataurl

Reply:
you also need to regenerate test_data after dropping this

Reply:
done and yep had to regen test_data